apps/hash-graph/lib/graph/src/shared/interval.rs

```diff
@@ -448,6 +451,45 @@ where
 {
 }
 
+impl<T, S, E> PartialOrd for Interval<T, S, E>
+where
+    T: PartialOrd,
+    S: IntervalBound<T>,
+    E: IntervalBound<T>,
+{
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        let start_ordering = compare_bounds(
+            self.start_bound(),
+            other.start_bound(),
+            BoundType::Start,
+            BoundType::Start,
+            PartialOrd::partial_cmp,
```

Comment:
Is `partial_cmp` only compatible with the return type because the `Option<Ordering>` return type satisfies `From<Ordering>`?

What happens here in case of None from the `partial_cmp` comparison?

Reply:
Yes, that's the reason why it's working. If `None` is returned from `partial_cmp`, `None` will be returned fro `compare_bounds` and this will be forwarded to the caller as well (`?` also passes `None` up)